### DISCUSSION TITLE
Async Redshifthook makes sync api calls.

### DISCUSSION THREAD

**Discussion by Mavtti** [astronomer/astronomer-providers#537] [Q&A]
Hello I wondered why in the `RedshiftDataHook` when running the async method `get_query_status`, the boto3 client is created in async while the api calls made with `client.describe_statement` are sync.

Why not make api calls async ?
Why not make client creation and api calls sync and only leave `asyncio.sleep` in async ?

https://github.com/astronomer/astronomer-providers/blob/102b24f7994628b213b97e499b890ab38a1b73ef/astronomer/providers/amazon/aws/hooks/redshift_data.py#L151-L160

**Comment by pankajkoti** [comment]
@Mavtti can you check if aiobotocore supports async calls for Redshift? In our check, we cannot find a way. And since the major part of the task goes on waiting for the query to finish we focused on awaiting there and the other parts which are quite fast, we did not focus much on making them async. We're open to ideas, discussions, solutions and contribution for the community. Feel free to raise a PR if you believe we can optimise parts of the code.

**Comment by pankajkoti** [comment]
Closing this discussion as it appears stale.
